scalafix-core/src/main/scala/scalafix/rewrite/NoXml.scala

```diff
@@ -0,0 +1,42 @@
+package scalafix
+package rewrite
+
+import scala.meta._
+import patch.TreePatch.AddGlobalImport
+
+case object NoXml extends Rewrite {
+
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    val patch = ctx.tree.tokens.collect {
+      case tok @ Token.Xml.Start() =>
+        ctx.addLeft(tok, "xml\"\"\"")
+
+      case tok @ Token.Xml.End() =>
+        ctx.addRight(tok, "\"\"\"")
+
+      case tok @ Token.Xml.SpliceStart() =>
+        ctx.addLeft(tok, "$")
+
+      case tok @ Token.Xml.Part(part) =>
+        if (part.contains('$'))
+          ctx.replaceToken(tok, part.replaceAllLiterally("$", "$$"))
+        else Patch.empty
+
+    }.asPatch
+
+    // Emit warning on `{{`
+    ctx.tree.tokens.foreach {
+      case tok @ Token.Xml.Part(part) if part.contains("{{") =>
+        val offset = part.indexOf("{{")
+        val pos =
+          Position.Range(tok.input, tok.start + offset, tok.start + offset + 2)
+        ctx.reporter.warn(
+          "Single opening braces within XML text don't need to be doubled",
```

Comment:
The issue is that there are many places in xml literals where braces don't need to be escaped:
```scala
scala> <a b="{{"/>
res0: scala.xml.Elem = <a b="{{"/>

scala> <![CDATA[{{]]> 
res1: scala.xml.PCData = <![CDATA[{{]]>

scala> <xml:unparsed>{{</xml:unparsed>
res2: scala.xml.Unparsed = {{

scala> <?foo {{?>
res3: scala.xml.ProcInstr = <?foo {{?>
```
We would need to parse the xml literal to implement this rewrite